src/Cli/func/Helpers/AzureHelper.cs

```diff
@@ -383,31 +383,49 @@ internal static async Task CheckFunctionHostStatusForFlex(Site functionApp, stri
             functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
             functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
 
+            var statusUri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+            ColoredConsole.WriteLine($"Target endpoint: {statusUri.GetLeftPart(UriPartial.Path)}");
+
             await RetryHelper.Retry(
                 async () =>
                 {
+                    // Remove the header first if it already exists from a previous retry
+                    if (functionAppReadyClient.DefaultRequestHeaders.Contains("x-ms-request-id"))
+                    {
+                        functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
+                    }
+
                     functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
-                    var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+
```

Comment:
Can also simplify:

```
functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
```

Remove handles the case where there is no request id to remove so this should be safe